chapters/lexicalstructure.tex

```diff
@@ -247,6 +247,44 @@ \subsection{Strings}\label{strings}
 \end{lstlisting}
 \end{nonnormative}
 
+\subsection{Units of Literal Constants}\label{units-literal-constants}
+
+Literal \lstinline!Real! and \lstinline!Integer! constants are implicitly assumed
+to have \lstinline!unit = "1"! if they appear in expressions containing variables,
+parameters, or constants that have a non-empty \lstinline!unit! attribute string.
+The inputs of transcendental built-in functions such as \lstinline!sin()!,
+\lstinline!cos()!, \lstinline!exp()!, etc., should have a dimensionless unit
+(e.g., "1" or "rad") if they involve expressions containing variables,
+parameters, or constants that have a non-empty unit attribute string; their
+outputs are implicitly assumed to have unit "1" if they are called within
+expressions containing variables, parameters, or constants that have a non-empty
+\lstinline!unit! attribute string.
```

Comment:
Let's begin by getting some of the boring stuff out of the way.

Please use sentence-based line breaking (also suggesting that unit string literals are formatted as inline code for consistency with the _Unit Expressions_ chapter):
```suggestion
Literal \lstinline!Real! and \lstinline!Integer! constants are implicitly assumed to have \lstinline!unit = "1"! if they appear in expressions containing variables, parameters, or constants that have a non-empty \lstinline!unit! attribute string.
The inputs of transcendental built-in functions such as \lstinline!sin()!, \lstinline!cos()!, \lstinline!exp()!, etc., should have a dimensionless unit (e.g., \lstinline!"1"! or \lstinline!"rad"!) if they involve expressions containing variables, parameters, or constants that have a non-empty unit attribute string; their outputs are implicitly assumed to have unit \lstinline!"1"! if they are called within expressions containing variables, parameters, or constants that have a non-empty \lstinline!unit! attribute string.
```

Reply:
Note that dimensionless should be written symmetrically for input and output, since we have both `sin` (from "rad" to "1") and its inverse `asin` (from "1" to "rad").

However, there are two exceptions - sqrt (which have separate rules), and atan2.

Reply:
(I suggest that we wait for @casella to get the sentence-based line breaking in place before we initiate further conversations attached to lines of code.)

Reply:
Reformatted to sentence-based line breaking in bdbea3b.

Reply:
> Note that dimensionless should be written symmetrically for input and output, since we have both `sin` (from "rad" to "1") and its inverse `asin` (from "1" to "rad").

See my comment [here](https://github.com/modelica/ModelicaSpecification/issues/2127#issuecomment-1271723893): from the point of view of dimensional consistency checking, "1" and "rad" are basically the same, i.e. dimensionless. Also, on can write `sin(L/R)` where `L` and `R` have unit "m" and that is perfectly fine, the input of `sin()` doesn't necessarily need to have unit "rad". What is important is to get the _dimension_ right, not the unit.

> However, there are two exceptions - sqrt (which have separate rules)

Sqrt is not a transcendental function 😃 

> and atan2.

Here we probably need to specify that the two inputs of atan2 need to have the same unit (if they have any). As to the output, see above about "1" and "rad" being the same for dimensional consistency checking.

I wanted to keep the new text short, of course we could put in a detailed list of how each of the 13 functions defined in Section 3.7.3 should be handled, but that would probably be a bit pedantic.

Reply:
> > Note that dimensionless should be written symmetrically for input and output, since we have both `sin` (from "rad" to "1") and its inverse `asin` (from "1" to "rad").
> 
> See my comment [here](https://github.com/modelica/ModelicaSpecification/issues/2127#issuecomment-1271723893): from the point of view of dimensional consistency checking, "1" and "rad" are basically the same, i.e. dimensionless. Also, on can write `sin(L/R)` where `L` and `R` have unit "m" and that is perfectly fine, the input of `sin()` doesn't necessarily need to have unit "rad". What is important is to get the _dimension_ right, not the unit.

Marking the new line with proposal.

> > However, there are two exceptions - sqrt (which have separate rules)
> 
> Sqrt is not a transcendental function 😃

?

> > and atan2.
> 
> Here we probably need to specify that the two inputs of atan2 need to have the same unit (if they have any). 

I would say that 'The function atan2(y, x) is from a unit-checking perspective treated as atan(y/x).'

Reply:
OK, I'll do that. I'm still a bit confused about all these github collaborative work features, sorry about that...

> sqrt is not a transcendental function

Hans, the sentence started with "The inputs of transcendental built-in functions...". The `sqrt()` function is not a [transcendental function](https://www.britannica.com/science/transcendental-function), so the sentence does not apply to it.

> I would say that 'The function atan2(y, x) is from a unit-checking perspective treated as atan(y/x).'

Sounds good, will do that when restarting the conversation.